manager/catalog_manager.go

```diff
@@ -347,13 +377,33 @@ func GetNewTemplateVersions(path string) (model.Template, bool) {
 		if ok {
 			log.Debugf("Template found by path: %s", path)
 			copyOfversionLinks := make(map[string]string)
+
 			for key, value := range templateMetadata.VersionLinks {
 				if value != path {
 					otherVersionTokens := strings.Split(value, ":")
 					oVersion := otherVersionTokens[2]
-					otherVersion, err := strconv.Atoi(oVersion)
 
-					if err == nil && otherVersion > currentVersion {
+					rancherComposePathOther := CatalogRootDir + catalogID + "/templates/" + parentPath + "/" + oVersion
```

Comment:
All this is repeated too, combine into a single function something like `getVersionForPath(path string) string {}`that reads a file, unmarshalls, and returns semversion.